resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala

```diff
@@ -334,22 +342,34 @@ private[spark] object KubernetesSparkRestServer {
   def main(args: Array[String]): Unit = {
     val parsedArguments = KubernetesSparkRestServerArguments.fromArgsArray(args)
     val secretFile = new File(parsedArguments.secretFile.get)
-    if (!secretFile.isFile) {
-      throw new IllegalArgumentException(s"Secret file specified by --secret-file" +
-        " is not a file, or does not exist.")
-    }
+    require(secretFile.isFile, "Secret file specified by --secret-file is not a file, or" +
+      " does not exist.")
     val sslOptions = if (parsedArguments.useSsl) {
-      val keyStorePassword = parsedArguments
-        .keyStorePasswordFile
-        .map(new File(_))
-        .map(Files.toString(_, Charsets.UTF_8))
       val keyPassword = parsedArguments
         .keyPasswordFile
         .map(new File(_))
         .map(Files.toString(_, Charsets.UTF_8))
+      val keyStorePassword = parsedArguments
+        .keyStorePasswordFile
+        .map(new File(_))
+        .map(Files.toString(_, Charsets.UTF_8))
+      validateSslOptions(parsedArguments)
+      val resolvedKeyStore = parsedArguments.keyStoreFile.map(new File(_)).orElse(
+        parsedArguments.keyPemFile.map(keyPemFile => {
+          parsedArguments.certPemFile.map(certPemFile => {
+            PemsToKeyStoreConverter.convertPemsToTempKeyStoreFile(
+              new File(keyPemFile),
+              new File(certPemFile),
+              "provided-key",
+              keyStorePassword,
+              keyPassword,
+              parsedArguments.keyStoreType)
+          })
+        }).getOrElse(throw new SparkException("When providing pem files, both the key and" +
+          " the certificate must be specified.")))
```

Comment:
mention in this message that these are for the listening port on the REST server